lib/Report/TimeDisconnectedSummary.php

```diff
@@ -386,6 +408,105 @@ public function getResults(SanitizerInterface $sanitizedParams)
 
             $rows[] = $entry;
+            $disconnectedDisplays[] = $entry['displayId'];
+        }
+
+        $displaySelect = '
+            SELECT display.display, display.displayId ';
+
+        if ($tags != '') {
+            $displaySelect .= ', (SELECT GROUP_CONCAT(DISTINCT tag)
+              FROM tag
+                INNER JOIN lktagdisplaygroup
+                  ON lktagdisplaygroup.tagId = tag.tagId
+                WHERE lktagdisplaygroup.displayGroupId = displaygroup.DisplayGroupID
+                GROUP BY lktagdisplaygroup.displayGroupId) AS tags ';
+        }
+
+        $displayBody = 'FROM `display` ';
+
+        if ($tags != '') {
+            $displayBody .= 'INNER JOIN `lkdisplaydg`
+                        ON lkdisplaydg.DisplayID = display.displayid
+                     INNER JOIN `displaygroup`
+                        ON displaygroup.displaygroupId = lkdisplaydg.displaygroupId
+                         AND `displaygroup`.isDisplaySpecific = 1 ';
+        }
+        $displayBody .= 'WHERE 1 = 1 ';
+
+        if (count($displayIds) > 0) {
+            $displayBody .= 'AND display.displayId IN (' . implode(',', $displayIds) . ') ';
+        }
+
+        if ($tags != '') {
+            if (trim($tags) === '--no-tag') {
+                $displayBody .= ' AND `displaygroup`.displaygroupId NOT IN (
+                    SELECT `lktagdisplaygroup`.displaygroupId
+                     FROM tag
+                        INNER JOIN `lktagdisplaygroup`
+                        ON `lktagdisplaygroup`.tagId = tag.tagId
+                    )
+                ';
+            } else {
+                $operator = $sanitizedParams->getCheckbox('exactTags') == 1 ? '=' : 'LIKE';
+
+                $displayBody .= " AND `displaygroup`.displaygroupId IN (
+                SELECT `lktagdisplaygroup`.displaygroupId
+                  FROM tag
+                    INNER JOIN `lktagdisplaygroup`
+                    ON `lktagdisplaygroup`.tagId = tag.tagId
+                ";
+                $i = 0;
+
+                foreach (explode(',', $tags) as $tag) {
+                    $i++;
+
+                    if ($i == 1) {
+                        $displayBody .= ' WHERE `tag` ' . $operator . ' :tags' . $i;
+                    } else {
+                        $displayBody .= ' OR `tag` ' . $operator . ' :tags' . $i;
+                    }
+
+                    if ($operator === '=') {
+                        $params['tags' . $i] = $tag;
+                    } else {
+                        $params['tags' . $i] = '%' . $tag . '%';
+                    }
+                }
+
+                $displayBody .= " ) ";
+            }
+        }
+
+        if ($onlyLoggedIn) {
+            $displayBody .= ' AND `display`.loggedIn = 1 ';
+        }
+
+        $displayBody .= '
+            GROUP BY display.display, display.displayId
+        ';
+
+        // Get a list of displays by filters
+        $displaySql = $displaySelect . $displayBody;
```

Comment:
Since we removed paging, we don't need to separate the displaySelect and displayBody - the whole query can just be displaySql.